### PR TITLE
Add ability to add JVM args to jobtype commands before they are launched.

### PR DESCRIPTION
**Pull request by djaiswal83**

### CODE REVIEW

File: az-core/src/main/java/azkaban/Constants.java [hunk 1]
```diff
@@ -153,6 +153,12 @@
   // Overridable plugin load properties
   public static final String AZ_PLUGIN_LOAD_OVERRIDE_PROPS = "azkaban.plugin.load.override.props";
 
+  // Append JVM args to job commands
+  public static final String AZ_JOB_COMMAND_ARGS = "azkaban.jvm.cmd.args";
```
Comment (jakhani) [inline az-core/src/main/java/azkaban/Constants.java:157]: Can you please change this to azkaban.jobs.java.opts?

File: az-core/src/main/java/azkaban/Constants.java [hunk 2]
```diff
@@ -153,6 +153,12 @@
...
+
+  // Ignore the above JVM args for jobtypes which have this true.
+  public static final String AZ_JOB_IGNORE_JVM_ARGS = "ignore.jvm.args";
```
Comment (jakhani) [inline az-core/src/main/java/azkaban/Constants.java:160]: Can you please change this to azkaban.jobs.ignore.java.opts?

File: az-core/src/main/java/azkaban/Constants.java [hunk 3]
```diff
@@ -153,6 +153,12 @@
   // Overridable plugin load properties
   public static final String AZ_PLUGIN_LOAD_OVERRIDE_PROPS = "azkaban.plugin.load.override.props";
 
+  // Append JVM args to job commands
```
Comment (jakhani) [inline az-core/src/main/java/azkaban/Constants.java:156]: Please correct the comment as well for Java options reference.

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java [hunk 4]
```diff
@@ -321,6 +324,27 @@ public void run() throws Exception {
...
 
+  /**
+   * This method fetches JVM args which maybe needed for jobtypes to function
+   * @param commands Original commands
+   * @return Updated commands.
+   */
+  private List<String> appendJVMArgs(List<String> commands) {
+    final String appendArgs = this.getSysProps().getString(
+        Constants.AZ_JOB_COMMAND_ARGS, null);
+    final boolean ignoreJVMArgs = this.getSysProps().getBoolean(
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java:335]: Please rename these variable according to property name change.

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java [hunk 5]
```diff
@@ -321,6 +324,27 @@ public void run() throws Exception {
...
+        Constants.AZ_JOB_IGNORE_JVM_ARGS, false);
+    if (ignoreJVMArgs || null == appendArgs) {
+      return commands;
+    }
+    // append the args to each command.
+    List<String> appendedCmds = new ArrayList<>(commands.size());
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java:341]: nit: final

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java [hunk 6]
```diff
@@ -321,6 +324,27 @@ public void run() throws Exception {
...
+   */
+  private List<String> appendJVMArgs(List<String> commands) {
+    final String appendArgs = this.getSysProps().getString(
+        Constants.AZ_JOB_COMMAND_ARGS, null);
+    final boolean ignoreJVMArgs = this.getSysProps().getBoolean(
+        Constants.AZ_JOB_IGNORE_JVM_ARGS, false);
+    if (ignoreJVMArgs || null == appendArgs) {
+      return commands;
+    }
+    // append the args to each command.
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java:340]: Please add logging for what you are appending to each command.

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java [hunk 7]
```diff
@@ -321,6 +324,29 @@ public void run() throws Exception {
     generateProperties(propFiles[1]);
   }
 
+  /**
+   * This method fetches JVM args which maybe needed for jobtypes to function
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java:328]: Nit: Java Options

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java [hunk 8]
```diff
@@ -262,6 +262,9 @@ public void run() throws Exception {
       }
     }
 
+    // Append any JVM args needed for job commands
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java:265]: Please rename appendJVMArgs. Also comments wherever you have mentioned JVM args.

Comment (djaiswal83) [reply azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java:265]: Ack

**Review by jakhani — APPROVED** [review]
Thank you for adding this fix.

**Pull request merged**